ibm/resource_ibm_cis_dns_record_test.go

```diff
@@ -0,0 +1,235 @@
+package ibm
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	v1 "github.com/IBM-Cloud/bluemix-go/api/cis/cisv1"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccIBMCISDNSRecord_Basic(t *testing.T) {
+	t.Parallel()
+	var record v1.DnsRecord
+	testName := "tf-acctest-basic"
+	resourceName := fmt.Sprintf("ibm_cis_dns_record.%s", testName)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		// Remove check destroy as this occurs after the CIS instance is deleted and fails with an auth error
+		//CheckDestroy: testAccCheckIBMCISDNSRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMCISDNSRecordConfigBasic(testName, cis_domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMCISDNSRecordExists(resourceName, &record),
+					resource.TestCheckResourceAttr(
+						resourceName, "name", testName+"."+cis_domain),
+					resource.TestCheckResourceAttr(
+						resourceName, "content", "192.168.0.10"),
+					resource.TestCheckResourceAttr(
+						resourceName, "data.%", "0"),
+					resource.TestMatchResourceAttr(
+						resourceName, "domain_id", regexp.MustCompile("^[a-z0-9]{32}$")),
+				),
+			},
+		},
+	})
+}
+
+func TestAccIBMCISDNSRecord_CaseInsensitive(t *testing.T) {
+	t.Parallel()
+	var record v1.DnsRecord
+	testName := "tf-acctest-case-insensitive"
+	resourceName := fmt.Sprintf("ibm_cis_dns_record.%s", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		// Remove check destroy as this occurs after the CIS instance is deleted and fails with an auth error
```

Comment:
May be instead of removing check destroy...You can add the required resources in each UAT

Reply:
Added comments. Delete of the resource is implicit in running the test successfully.